relayer/connection-tx.go

```diff
@@ -324,3 +328,26 @@ func InitializeConnection(src, dst *Chain, srcUpdateHeader, dstUpdateHeader *tmc
 		return false, true, fmt.Errorf("connection ends already created")
 	}
 }
+
+// FindMatchingConnection will determine if there already exists a unintialized connection between source and counterparty.
+func FindMatchingConnection(source, counterparty *Chain) (string, bool) {
+	// TODO: add appropriate offset and limits, along with retries
+	connectionsResp, err := source.QueryConnections(0, 1000)
+	if err != nil {
+		if source.debug {
+			source.Log(fmt.Sprintf("Error: querying connections on %s failed: %v", source.PathEnd.ChainID, err))
+		}
+		return "", false
+	}
+
+	for _, connection := range connectionsResp.Connections {
+		if connection.ClientId == source.PathEnd.ClientID &&
```

Comment:
We need to ensure that the:
- Client ID matches the one in the config
- counterparty client id is counterparty.PathEnd.ClientID
- The versions matches the ones we use
- The DelayPeriod matches the one we use
- The counterparty prefix matches what we expect

Extra logic can be done based on the current state. If the state is OPEN, the counterparty connection id must match counterparty.PathEnd.ConnectionID

There is some more advanced version checking that could occur in INIT and TRYOPEN states but this is trying to mirror the negotiation that happens on chain and shouldn't be an issue at all until ibcv2 so I think we could just open an issue for now and ignore it

The connection struct evolves throughout the handshake so the general logic is to figure out what state it is in and then ensure it matches our expected parameters for that state, with OPEN requiring the most matching parameters